src/server/jenkinsApi.js

```diff
@@ -2,130 +2,166 @@ const express = require('express');
 router.get('/', (req, res) => {
-  res.send('jenkins api root');
+    res.send('jenkins api root');
 });
 
 // Get build information from the main jenkins job
 router.get('/main/job', (req, res, next) => {
-  let url = MAIN_JOB_URL_API;
-  console.log('Getting ' + url);
-  axios.get(url)
-    .then(result => {
-      res.status(200).json(result.data);
-    })
-    .catch(error => {
-      next(error)
+    let url = MAIN_JOB_URL_API;
+    console.log('Getting ' + url);
+    axios.get(url).then(result => {
+        res.status(200).json(result.data);
+    }).catch(error => {
+        next(error)
     });
 });
 
 // Get build information for a specific build number
 router.get('/main/build/:number', (req, res, next) => {
-  let url = MAIN_BASE_URL + '/' + req.params.number + JSON_API;
-  console.log('Getting ' + url);
-  axios.get(url)
-    .then(result => {
-      res.status(200).json(result.data);
-    })
-    .catch(error => {
-      next(error)
+    let url = MAIN_BASE_URL + req.params.number + '/' + JSON_API;
+    console.log('Getting ' + url);
+    axios.get(url).then(result => {
+        res.status(200).json(result.data);
+    }).catch(error => {
+        next(error)
     });
 });
 
 // Get test results from latest main run
 router.get('/main/test_report/latest', (req, res, next) => {
-  let url = MAIN_LAST_SUCCESSFUL_TEST_REPORT_URL_API;
-  console.log('Getting ' + url);
-  axios.get(url)
-    .then(result => {
-      let data = result.data;
-      data.url = MAIN_LAST_SUCCESSFUL_TEST_REPORT_URL;
-
-      res.status(200).json(data);
-    })
-    .catch(error => {
-      next(error)
+    let url = MAIN_LAST_SUCCESSFUL_TEST_REPORT_URL_API;
+    console.log('Getting ' + url);
+    axios.get(url).then(result => {
+        let data = result.data;
+        data.url = MAIN_LAST_SUCCESSFUL_TEST_REPORT_URL;
+
+        res.status(200).json(data);
+    }).catch(error => {
+        next(error)
     });
 });
 
 // Get test results from specific main run
 router.get('/main/test_report/:buildNumber', (req, res, next) => {
-  let url = MAIN_BASE_URL + '/' + req.params.buildNumber + '/testReport' + JSON_API
-  console.log('Getting ' + url);
-  axios.get(url)
-    .then(result => {
-      res.status(200).json(result.data);
-    })
-    .catch(error => {
-     next(error);
+    let url = MAIN_BASE_URL + req.params.buildNumber + '/testReport/' + JSON_API
+    console.log('Getting ' + url);
+    axios.get(url).then(result => {
+        res.status(200).json(result.data);
+    }).catch(error => {
+        next(error);
     });
 });
 
 // Get build information from the qa jenkins job
 router.get('/qa/job', (req, res, next) => {
-  let url = QA_JOB_URL_API;
-  console.log('Getting ' + url);
-  axios.get(url)
-    .then(result => {
-      res.status(200).json(result.data);
+    getQAJobUrl().then(() => {
+        let url = QA_JOB_URL_API;
+        console.log('Getting ' + url);
+        return url;
+    }).then( (url) => {
+        axios.get(url).then(result => {
```

Comment:
you should do
return axios.get(url)

and then your next then can be moved out instead of nesting promises chains.  Let me know if you need a discussion on promises.